pkg/controller/apicast/apicast_logic_reconciler.go

```diff
@@ -418,41 +421,45 @@ func (r *APIcastLogicReconciler) reconcileDeployment(desiredDeployment appsv1.De
 	err := r.Client().Get(context.TODO(), r.namespacedName(&desiredDeployment), &existingDeployment)
 	if err != nil {
 		if errors.IsNotFound(err) {
+			r.Logger().Info(fmt.Sprintf("Creating %s", k8sutils.ObjectInfo(&desiredDeployment)))
 			err = r.Client().Create(context.TODO(), &desiredDeployment)
-			r.Logger().Info("Creating Deployment...")
 			return err
 		}
 		return err
 	}
 
 	changed := false
 
-	if !reflect.DeepEqual(existingDeployment.Spec.Replicas, desiredDeployment.Spec.Replicas) {
+	if existingDeployment.Spec.Replicas != desiredDeployment.Spec.Replicas {
 		existingDeployment.Spec.Replicas = desiredDeployment.Spec.Replicas
 		changed = true
 	}
-	if !reflect.DeepEqual(existingDeployment.Spec.Template.Spec.Containers[0].Image, desiredDeployment.Spec.Template.Spec.Containers[0].Image) {
+	if existingDeployment.Spec.Template.Spec.Containers[0].Image != desiredDeployment.Spec.Template.Spec.Containers[0].Image {
 		existingDeployment.Spec.Template.Spec.Containers[0].Image = desiredDeployment.Spec.Template.Spec.Containers[0].Image
 		changed = true
 
 	}
-	if !reflect.DeepEqual(existingDeployment.Spec.Template.Spec.ServiceAccountName, desiredDeployment.Spec.Template.Spec.ServiceAccountName) {
+	if existingDeployment.Spec.Template.Spec.ServiceAccountName != desiredDeployment.Spec.Template.Spec.ServiceAccountName {
 		changed = true
 		existingDeployment.Spec.Template.Spec.ServiceAccountName = desiredDeployment.Spec.Template.Spec.ServiceAccountName
 	}
 
-	if !reflect.DeepEqual(existingDeployment.Spec.Template.Spec.Containers[0].Env, desiredDeployment.Spec.Template.Spec.Containers[0].Env) {
-		changed = true
-		existingDeployment.Spec.Template.Spec.Containers[0].Env = desiredDeployment.Spec.Template.Spec.Containers[0].Env
-	}
+	updatedTmp := ReconcileEnvVar(&existingDeployment.Spec.Template.Spec.Containers[0].Env, desiredDeployment.Spec.Template.Spec.Containers[0].Env)
+	changed = changed || updatedTmp
 
-	// TODO should we merge instead of set the desired annotations? Keep in mind that this is are annotations
-	// of the PodTemplate, not the Pod itself. Can a PodTemplate be modified by other elements?
+	// They are annotations of the PodTemplate, part of the Spec, not part of the meta info of the Pod or Environment object itself
+	// It is not expected any controller to update them, so we use "set" approach, instead of merge.
+	// This way any removed annotation from desired (due to change in CR) will be removed in existing too.
 	if !reflect.DeepEqual(existingDeployment.Spec.Template.Annotations, desiredDeployment.Spec.Template.Annotations) {
 		changed = true
 		existingDeployment.Spec.Template.Annotations = desiredDeployment.Spec.Template.Annotations
 	}
 
+	if !reflect.DeepEqual(existingDeployment.Spec.Template.Labels, desiredDeployment.Spec.Template.Labels) {
```

Comment:
Why are we reconciling Deployment's PodTemplate labels?

If I'm not wrong, we are not providing configuration of them through the CR.

Reply:
`apicastFullName := "apicast-" + r.APIcastCR.Name`

Reply:
I guess we do

Reply:
You cannot change a CR name once deployed so no point in doing so

Reply:
why not?? then why it is exposed?

Reply:
It's not exposed. All object types have a name

Reply:
You cannot change a Kubernetes object name once it is deployed.

Reply:
right, no label reconciliation then